webvh/webvh/protocols/endorse_attested_resource/tests/test_protocol.py

```diff
@@ -0,0 +1,113 @@
+            BaseResponder, mock.AsyncMock(BaseResponder, autospec=True)
+        )
+        self.profile.context.injector.bind_instance(EventBus, EventBus())
+        self.profile.context.injector.bind_instance(KeyTypes, KeyTypes())
+        self.profile.context.injector.bind_instance(DIDResolver, TEST_RESOLVER)
+        self.request = mock.MagicMock(
+            app={},
+            match_info={},
+            query={},
+            __getitem__=lambda _, k: self.request_dict[k],
+        )
+        async with self.profile.session() as session:
+            # Create known witness key for test server
+            await MultikeyManager(session).create(
+                alg="ed25519",
+                kid=f"webvh:{TEST_DOMAIN}@witnessKey",
+                # seed=TEST_WITNESS_SEED,
+            )
+
+    async def test_record(self):
+        await record.set_pending_scid(self.profile, TEST_SCID)
+        await record.get_pending_scids(self.profile)
+        await record.remove_pending_scid(self.profile, TEST_SCID)
+        await record.get_pending_scids(self.profile)
+
+        await record.save_pending_record(self.profile, TEST_SCID, TEST_RECORD)
+        await record.get_pending_records(self.profile)
+        await record.get_pending_record(self.profile, TEST_SCID)
+        await record.remove_pending_record(self.profile, TEST_SCID)
+
+        with self.assertRaises(AttributeError):
+            await record.get_pending_records(self.profile)
+            await record.get_pending_record(self.profile, TEST_SCID)
+
+    @mock.patch(
+        "aiohttp.ClientSession.post",
+        mock.AsyncMock(
+            return_value=mock.MagicMock(
+                json=mock.AsyncMock(
+                    return_value={
+                        "id": TEST_RESOURCE_ID,
+                        "content": {},
+                        "proof": {},
+                    }
+                )
+            )
+        ),
+    )
+    async def test_handler(self):
+        self.profile.settings.set_value(
+            "plugin_config",
+            {"did-webvh": {"server_url": "https://example.com", "auto_attest": False}},
+        )
+        context = RequestContext(self.profile)
+        context.message = WitnessRequest(document=TEST_RECORD)
+        context.connection_record = ConnRecord(
+            alias=f"webvh:{TEST_DOMAIN}@witness",
+            state="active",
+            connection_id="123",
+        )
+        await WitnessRequestHandler().handle(
+            context, mock.AsyncMock(BaseResponder, autospec=True)
+        )
+
+        witness_proof = {"type": "DataIntegrityProof"}
+        context = RequestContext(self.profile)
+        context.message = WitnessResponse(
+            state=WitnessingState.SUCCESS.value,
+            document=TEST_RECORD,
+            witness_proof=witness_proof,
+        )
+        await WitnessResponseHandler().handle(
+            context, mock.AsyncMock(BaseResponder, autospec=True)
+        )
```

Comment:
Is there anything we could assert on to ensure the test is actually doing what we think it is doing?

Reply:
returned a status from the handler and added the test assertion